context/app/static/js/shared-styles/charts/utils.js

```diff
@@ -5,4 +5,11 @@ function getChartDimensions(parentWidth, parentHeight, margin) {
   return { xWidth, yHeight };
 }
 
-export { getChartDimensions };
+function trimStringWithMiddleEllipsis(str) {
+  if (str.length > 25) {
+    return `${str.substr(0, 15)}...${str.substr(str.length - 10, str.length)}`;
+  }
+  return str;
```

Comment:
Regex might be good here?

```suggestion
  return str.replace(/^(.{15}).+(.{10})$/,'$1...$2');
```